tools/wt_binary_decode.py

```diff
@@ -0,0 +1,652 @@
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+# Dump/decode a WiredTiger .wt file.
+
+# This is not a complete dump program and should not be depended upon for correctness --
+# see "wt dump" for that.  But this is standalone (doesn't require linkage with any WT
+# libraries), and may be useful as 1) a learning tool 2) quick way to hack/extend dumping.
+
+import binascii, codecs, io, os, string, sys
+
+decode_version = "2023-03-03.0"
+
+# A container for fields in the WT_PAGE_HEADER
+class PageHeader:
+    pass
+
+# A container for fields in the WT_BLOCK_HEADER
+class BlockHeader:
+    pass
+
+################################################################
+# Borrowed from intpacking.py, with small adjustments.
+# Variable-length integer packing
+# need: up to 64 bits, both signed and unsigned
+#
+# Try hard for small values (up to ~2 bytes), after that, just encode the
+# length in the first byte.
+#
+#  First byte | Next |                        |
+#  byte       | bytes| Min Value              | Max Value
+# ------------+------+------------------------+--------------------------------
+# [00 00xxxx] | free | N/A                    | N/A
+# [00 01llll] | 8-l  | -2^64                  | -2^13 - 2^6
+# [00 1xxxxx] | 1    | -2^13 - 2^6            | -2^6 - 1 
+# [01 xxxxxx] | 0    | -2^6                   | -1
+# [10 xxxxxx] | 0    | 0                      | 2^6 - 1 
+# [11 0xxxxx] | 1    | 2^6                    | 2^13 + 2^6 - 1
+# [11 10llll] | l    | 2^13 + 2^6             | 2^64 - 1
+# [11 11xxxx] | free | N/A                    | N/A
+
+NEG_MULTI_MARKER = 0x10
+NEG_2BYTE_MARKER = 0x20
+NEG_1BYTE_MARKER = 0x40
+POS_1BYTE_MARKER = 0x80
+POS_2BYTE_MARKER = 0xc0
+POS_MULTI_MARKER = 0xe0
+
+NEG_1BYTE_MIN = -2**6
+NEG_2BYTE_MIN = -2**13 + NEG_1BYTE_MIN
+POS_1BYTE_MAX = 2**6 - 1
+POS_2BYTE_MAX = 2**13 + POS_1BYTE_MAX
+
+MINUS_BIT = -1 << 64
```

Comment:
I don't see this constant being used. Are you sure it's supposed to be shifted by 64 bits and not 63?